skills/csharp/experimental/musicskill/Dialogs/MainDialog.cs

```diff
@@ -313,23 +313,19 @@ private async Task<DialogTurnResult> FinalStepAsync(WaterfallStepContext stepCon
             var state = await _stateAccessor.GetAsync(stepContext.Context, () => new SkillState());
             if (stepContext.Context.IsSkill())
             {
-                // EndOfConversation activity should be passed back to indicate that VA should resume control of the conversation
-                var endOfConversation = new Activity(ActivityTypes.EndOfConversation)
-                {
-                    Code = EndOfConversationCodes.CompletedSuccessfully,
-                    Value = stepContext.Result,
-                };
+                var result = stepContext.Result;
 
-                if (state.IsAction && stepContext.Result == null)
+                if (state.IsAction && result as ActionResult == null)
                 {
-                    endOfConversation.Value = new ActionResult() { ActionSuccess = false };
+                    result = new ActionResult() { ActionSuccess = false };
                 }
 
-                await stepContext.Context.SendActivityAsync(endOfConversation, cancellationToken);
-                return await stepContext.EndDialogAsync();
+                state.Clear();
```

Comment:
I added state.clear() for the skills that seem to be using state in FinalStepAsync just so that state is cleared after every conversation. Please let me know if there's problem with this
@bobokids  @xieofxie  @xhr0804 @KayMKM

Reply:
I have added state.clear() at the beginning of RouteStep to make sure each dialog start with a cleared state. So i think either clear in FinalStep or not is OK.

Reply:
No, I don't think it is necessary. Skills have been cleared somewhere and they may not follow same flow pattern